common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -4036,6 +4039,15 @@ object CelebornConf extends Logging {
       .bytesConf(ByteUnit.BYTE)
       .createWithDefaultString("64k")
 
+  val PARTITION_READER_CHECKPOINT_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.client.partition.reader.checkpoint.enabled")
+      .categories("client")
+      .version("0.6.0")
+      .doc("Whether or not checkpoint reads when re-creating a partition reader. Setting to true minimizes" +
+        " the amount of unnecessary reads during partition read retries")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
We can consider enabling it by default, as it doesn't seem to introduce much overhead. WDYT？

Reply:
I think enabling by default would be a good idea, however in 0.6, it's currently disabled by default. I'd like to ensure parity across client version. WDYT?

Reply:
> I think enabling by default would be a good idea, however in 0.6, it's currently disabled by default. I'd like to ensure parity across client version. WDYT?

Thanks, I am okay with enabling this in 0.6

Reply:
Sounds good. I've default enabled here. I'll raise a followup to enable in 0.6